jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -980,4 +979,29 @@ public String getExcludedMethodList()
     {
         return String.join(",", getExcludedMethods());
     }
+
+    /**
+     * Set the {@link DeflaterPool} that is used to manage instances of {@link Deflater}
+     * @param deflaterPool
+     */
+    public void setDeflaterPool(DeflaterPool deflaterPool)
+    {
+        if(isStarted())
+            throw new IllegalStateException(getState());
+
+        _deflaterPool = deflaterPool;
+    }
+
+    public int getDeflaterPoolCapacity()
+    {
+        return POOL_CAPACITY;
+    }
+
+    public void setDeflaterPoolCapacity(int capacity)
+    {
+        if(isStarted())
+            throw new IllegalStateException(getState());
+
+        POOL_CAPACITY = capacity;
+    }
```

Comment:
Having both `setDeflaterPool()` and `setDeflaterPoolCapacity()` seems a duplication to me, and potential for confusion: if I set my own `DeflaterPool`, what effect has to call `setDeflaterPoolCapacity()` afterwards?
What if it is called before?

I would remove `setDeflaterPoolCapacity()` because it's an implementation detail that does not need to be exposed in  `GzipHandler`. If you want a custom size for the deflater pool, just create your own and call `setDeflaterPool()`.

Reply:
I agree that both is not necessary.  But I'd prefer a setter for capacity, mostly because that will be more often used and easy to do in XML etc.

Instead of `setDeflaterPool`  have a protected `newDeflaterPool(int capacity)` method that users can override if they want.